tests/test_init.py

```diff
@@ -43,3 +36,8 @@ def test_import_error_invalid_attr():
     # and may vary between Python versions.
     with pytest.raises(ImportError):
         from propcache import invalid_attr  # noqa: F401
+
+
+def test_no_wildcard_imports():
+    """Verify wildcard imports are prohibited."""
+    assert not propcache.__all__
```

Comment:
We can't clearly explicitly test wildcard imports since `SyntaxError: import * only allowed at module level` but making sure __all__ is empty is probably enough

Reply:
Oh, I didn't realize wildcards aren't allowed in functions. TIL! Clearly, I never tried that :)
One way could be importing in a subprocess. But what you did is more elegant and a subprocess might be an overkill, I suppose.